docs/usage/visualize.ipynb

```diff
@@ -539,6 +533,66 @@
     "graphviz.Source(dot)"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Collapsing the graph also works for custom properties:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": [
+     "scroll-input"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "from fractions import Fraction\n",
+    "\n",
+    "\n",
+    "def render_jpc_ig(state: State) -> str:\n",
+    "    particle = state.particle\n",
+    "    text = render_fraction(particle.spin)\n",
+    "    if particle.parity is not None:\n",
+    "        text += render_sign(particle.parity)\n",
+    "    if particle.c_parity is not None:\n",
+    "        text += render_sign(particle.c_parity)\n",
+    "    if particle.isospin is not None:\n",
+    "        text += f\"({render_fraction(particle.isospin.magnitude)}\"  # with opening brace\n",
+    "        if particle.g_parity is not None:\n",
+    "            text += f\"{render_sign(particle.g_parity)})\"  # with closing brace\n",
```

Comment:
I would combine the `if`: only append $I^G$ if both `isospin` and `g_parity` exist.